hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -164,6 +199,59 @@ def _default_employee(self):
         compute='_compute_total_time',
         store=True,
     )
+    possible_reviewer_ids = fields.Many2many(
+        string='Possible Reviewers',
+        comodel_name='res.users',
+        compute='_compute_possible_reviewer_ids',
+        store=True,
+    )
+    can_review = fields.Boolean(
+        string='Can Review',
+        compute='_compute_can_review',
+        search='_search_can_review',
+    )
+    complete_name = fields.Char(
+        string='Complete Name',
+        compute='_compute_complete_name',
+    )
+
+    @api.model
+    def fields_view_get(self,
+                        view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        res = super().fields_view_get(
+            view_id=view_id,
+            view_type=view_type,
+            toolbar=toolbar,
+            submenu=submenu,
+        )
+
+        view = etree.XML(res['arch'])
+        self._modify_view_for_policy(view_type, view)
+        res['arch'] = etree.tostring(
+            view,
+            encoding='unicode',
+        ).replace('\t', '')
+
+        return res
+
+    @api.model
+    def _modify_view_for_policy(self, view_type, view):
+        """ Hook for extensions """
+        review_policy = self.env.user.company_id.timesheet_sheet_review_policy
+        field = None
+        if review_policy == 'project_manager' and view_type == 'tree':
+            field = view.find(".//field[@name='project_id']")
+        elif review_policy == 'department_manager' and view_type == 'tree':
+            field = view.find(".//field[@name='department_id']")
+
+        if field is not None:
+            field.set('invisible', '0')
```

Comment:
Shouldn't all this black magic be moved to a separate custom module? I see it's made just for the visibility of two fields in the tree view. I think it doesn't worth it.

Reply:
But it's the only way to make those columns conditionally invisible, and leaving them as is would only pollute the UI

Reply:
Module `hr_timesheet_sheet` should be a base module for being extended (and thanks for splitting the methods to hook methods) but this kind of changes are going a bit too far IMO. I would prefer to keep the module as a basis, although it became already quite complex compared to the original V10 version.

Regarding this particular part of code, are you sure that the conditionally invisible fields are necessary? The original tree view doesn't have those two fields and therefore it's open for being customized.

Reply:
Well, those fields were added for `department_manager` validation policy, so for all others these fields are not valuable to be seen

Reply:
As I said, I think it doesn't worth it.

Reply:
Does not worth adding `department_manager` policy? Does not worth hiding policy-specific columns?

Reply:
Not by adding such black magic in this module.

I would love to continue to discuss about this, but my holidays are now starting.

Reply:
:) It's not black magic, it's even used in Odoo itself :P Please ping me when you're back from holidays